execute/row_fn.go

```diff
@@ -286,29 +290,35 @@ func (f *RowMapFn) Eval(ctx context.Context, row int, cr flux.ColReader) (values
 }
 
 type RowReduceFn struct {
-	rowFn
+	dynamicFn
 }
 
-func NewRowReduceFn(fn *semantic.FunctionExpression, scope compiler.Scope) (*RowReduceFn, error) {
-	r, err := newRowFn(fn, scope)
+func NewRowReduceFn(fn *semantic.FunctionExpression, scope compiler.Scope) *RowReduceFn {
+	return &RowReduceFn{
+		dynamicFn: newDynamicFn(fn, scope),
+	}
+}
```

Comment:
I'm guessing that not propagating the error here has something to do with no longer keeping the additional state on `dynamicFn`, but I'm having a hard time understanding what. Also, does this just no longer error?

Reply:
When I was refactoring this I found all of these functions never errored for some reason. If I was going to make a breaking change I may as well fix the signatures at the same time.